pkg/broker/filter/filter_handler_test.go

```diff
@@ -103,46 +106,54 @@ func TestReceiver(t *testing.T) {
 		},
 		"Trigger doesn't have SubscriberURI": {
 			triggers: []*eventingv1.Trigger{
-				makeTriggerWithoutSubscriberURI(),
+				makeTrigger(withoutSubscriberURI()),
 			},
 			expectedStatus:     http.StatusBadRequest,
 			expectedEventCount: true,
 		},
 		"Trigger without a Filter": {
 			triggers: []*eventingv1.Trigger{
-				makeTriggerWithoutFilter(),
+				makeTrigger(),
 			},
 			expectedDispatch:          true,
 			expectedEventCount:        true,
 			expectedEventDispatchTime: true,
 		},
 		"No TTL": {
 			triggers: []*eventingv1.Trigger{
-				makeTrigger(makeTriggerFilterWithAttributes("some-other-type", "")),
+				makeTrigger(withAttributesFilter(&eventingv1.TriggerFilter{
+					Attributes: map[string]string{"type": "some-other-type"},
+				})),
 			},
 			event: makeEventWithoutTTL(),
 		},
 		"Wrong type": {
 			triggers: []*eventingv1.Trigger{
-				makeTrigger(makeTriggerFilterWithAttributes("some-other-type", "")),
+				makeTrigger(withAttributesFilter(&eventingv1.TriggerFilter{
+					Attributes: map[string]string{"type": "some-other-type"},
+				})),
 			},
 			expectedEventCount: false,
 		},
 		"Wrong source": {
 			triggers: []*eventingv1.Trigger{
-				makeTrigger(makeTriggerFilterWithAttributes("", "some-other-source")),
+				makeTrigger(withAttributesFilter(&eventingv1.TriggerFilter{
+					Attributes: map[string]string{"source": "some-other-source"},
+				})),
 			},
 			expectedEventCount: false,
 		},
 		"Wrong extension": {
 			triggers: []*eventingv1.Trigger{
-				makeTrigger(makeTriggerFilterWithAttributesAndExtension("", "", "some-other-extension")),
+				makeTrigger(withAttributesFilter(&eventingv1.TriggerFilter{
+					Attributes: map[string]string{extensionName: "some-other-extension"},
+				})),
 			},
 			expectedEventCount: false,
 		},
 		"Dispatch failed": {
 			triggers: []*eventingv1.Trigger{
-				makeTrigger(makeTriggerFilterWithAttributes("", "")),
+				makeTrigger(withAttributesFilter(&eventingv1.TriggerFilter{})),
```

Comment:
nit: there are a number of `makeTrigger(withAttributesFilter(&eventingv1.TriggerFilter{}))` in this test function that might be superfluous. It looks like `makeTrigger` does not add any filter by default, hence the empty trigger is not needed to be added.

```suggestion
				makeTrigger(),
```

Unless you want to highlight that no filters are being configured in which case disregard my comment.

Reply:
I think it's useful to explicitly show in the test case there are no filters indeed. But I get your point.